### PR TITLE
Tree Validation

### PR DESCRIPTION
**Pull request by awildturtok**

### CODE REVIEW

File: backend/src/main/java/com/bakdata/conquery/models/concepts/tree/Prefix.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+	@JsonValue
+	@NotNull
+	private final String value;
+
+	@Override
+	public int compareTo(@NotNull Prefix o) {
+		final int min = Math.min(getValue().length(), o.getValue().length());
+
+		return getValue().substring(0, min).compareToIgnoreCase(o.getValue().substring(0,min));
```
Comment (thoniTUB) [inline backend/src/main/java/com/bakdata/conquery/models/concepts/tree/Prefix.java:23]: Das ignoreCase ist  glaube ich ein Problem, weil A-z alle Buchstaben abdecken würde

Comment (thoniTUB) [reply backend/src/main/java/com/bakdata/conquery/models/concepts/tree/Prefix.java:23]: Aber ich bin mir auch nicht sicher

File: backend/src/main/java/com/bakdata/conquery/models/concepts/tree/ConceptTreeConnector.java [hunk 2]
```diff
@@ -40,7 +40,7 @@
 	@NsIdRef @CheckForNull
 	private Column column = null;
 
-	private CTCondition condition = null;
+	private ConceptTreeCondition condition = null;
```
Comment (thoniTUB) [inline backend/src/main/java/com/bakdata/conquery/models/concepts/tree/ConceptTreeConnector.java:43]: Brauch der auch noch eine Validierung?

File: backend/src/test/java/com/bakdata/conquery/models/concepts/tree/ConceptTreeChildTest.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import com.bakdata.conquery.models.concepts.conditions.ColumnEqualCondition;
+import com.bakdata.conquery.models.concepts.conditions.PrefixCondition;
+import com.bakdata.conquery.models.concepts.conditions.PrefixRangeCondition;
+import org.junit.jupiter.api.Test;
+
+class ConceptTreeChildTest {
+	@Test
+	public void encloses() {
+		final ConceptTreeChild root = new ConceptTreeChild();
+		root.setCondition(new PrefixRangeCondition("A00", "A10"));
```
Comment (thoniTUB) [inline backend/src/test/java/com/bakdata/conquery/models/concepts/tree/ConceptTreeChildTest.java:17]: Kannst du hier bitte auch den häufigen Edge-Case hinzufügen, dass die Kinder den Parent komplett ausfüllen?

Comment (awildturtok) [reply backend/src/test/java/com/bakdata/conquery/models/concepts/tree/ConceptTreeChildTest.java:17]: häufiger edge case 👀
